erpnext/projects/report/profitability/profitability.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+
+def execute(filters=None):
+	columns, data = [], []
+	data = get_data(filters)
+	columns = get_columns()
+	charts = get_chart_data(data)
+	return columns, data, None, charts
+
+def get_data(filters):
+	conditions = get_conditions(filters)
+	default_working_hours = frappe.db.get_single_value("HR Settings", "default_working_hours")
+	sql = """ 
+			select 
+				*, 
+				t.gross_pay * t.utilization as fractional_cost, 
+				t.grand_total - t.gross_pay * t.utilization as profit
```

Comment:
Let's separate out the calculations in a separate method. Always better to do so when the calculation can change based on some conditions. 

For example:
- If the employee currency is not the same as the company currency, you will have to take into consideration `base_gross_pay` instead of `gross_pay`.
- Similarly, if sales invoice currency is not the same as company currency, you will have to consider `base_grand_total` instead of `grand_total`

So, just fetch the data in this function and then calculate the metrics by handling these cases in another function. Also, write multiline SQL with better formatting. Having conditions on separate lines, writing keywords in all caps makes it easier to read.

![image](https://user-images.githubusercontent.com/24353136/114655389-c2005400-9d09-11eb-91be-61f7422b42a8.png)

![image](https://user-images.githubusercontent.com/24353136/114655438-d5abba80-9d09-11eb-9fa9-525e49330ee3.png)